File: apps/frontend/src/lib/contracts/assessment.ts
export type AssessmentDomain =
  | "ui_ux"
  | "compliance_security"
  | "backend_functionality"
  | "client_contact_simulation"
  | "efficiency_workflow"
  | "website_portal"

export type FindingSeverity = "P0" | "P1" | "P2" | "P3"
export type FindingStatus = "OPEN" | "IN_PROGRESS" | "CLOSED"
export type RetestResult = "PASS" | "FAIL"

export interface AssessmentRetest {
  result: RetestResult
  evidence: string[]
  recorded_at: string
}

export interface AssessmentFinding {
  finding_id: string
  domain: AssessmentDomain
  title: string
  severity: FindingSeverity
  owner: string
  status: FindingStatus
  notes: string
  retests: AssessmentRetest[]
  created_at: string
  updated_at: string
}

const DOMAIN_LABELS: Record<AssessmentDomain, string> = {
  ui_ux: "UI and UX",
  compliance_security: "Compliance and Security",
  backend_functionality: "Backend Functionality",
  client_contact_simulation: "Client/Contact Simulation",
  efficiency_workflow: "Efficiency and Workflow",
  website_portal: "Website and Portal",
}

export function toDomainLabel(domain: AssessmentDomain): string {
  return DOMAIN_LABELS[domain]
}

File: apps/backend/app/models/assessment.py
from datetime import UTC, datetime
from typing import Literal

from pydantic import BaseModel, Field


DomainName = Literal[
    "ui_ux",
    "compliance_security",
    "backend_functionality",
    "client_contact_simulation",
    "efficiency_workflow",
    "website_portal",
]

Severity = Literal["P0", "P1", "P2", "P3"]
FindingStatus = Literal["OPEN", "IN_PROGRESS", "CLOSED"]
RetestResult = Literal["PASS", "FAIL"]


def utc_now_iso() -> str:
    return datetime.now(UTC).isoformat().replace("+00:00", "Z")


class RetestRecord(BaseModel):
    result: RetestResult
    evidence: list[str] = Field(default_factory=list)
    recorded_at: str = Field(default_factory=utc_now_iso)


class Finding(BaseModel):
    finding_id: str
    domain: DomainName
    title: str
    severity: Severity
    owner: str
    status: FindingStatus = "OPEN"
    notes: str = ""
    retests: list[RetestRecord] = Field(default_factory=list)
    created_at: str = Field(default_factory=utc_now_iso)
    updated_at: str = Field(default_factory=utc_now_iso)
